measureme/src/event_id.rs

```diff
@@ -1,20 +1,24 @@
 use crate::{Profiler, SerializationSink, StringComponent, StringId};
 
-/// Event IDs are strings conforming to the following grammar:
+/// Event IDs are strings conforming to the following (eBNF) grammar:
 ///
 /// ```ignore
 ///   <event_id> = <label> {<argument>}
 ///   <label> = <text>
-///   <argument> = '\x1E' <text>
+///   <argument> = ['\x1D' <argument_name>] '\x1E' <argument_value>
+///   <argument_name> = <text>
+///   <argument_value> = <text>
```

Comment:
I wonder if it would be simpler to make the grammar:

```
<argument> = '\x1E' (<argument_name_value_pair> | <argument_value>)
<argument_name_value_pair> = <text> '\x1D' <text>
<argument_value> = <text>
```

Then we could always parse the leading `\x1E` and then do a string split on the `\x1D` if it exists.

Reply:
I think that would also make the format compatible with the current format. It's fine if we need to break that but we'll have to bump the format number and coordinate redeploying new versions on perf.rlo.

Reply:
It sounds like there might be another breaking change in the works from another contributor so if you think it's simpler to use the current strategy rather than this suggestion, that's fine with me!